packages/spotlight/Spottable/Spottable.js

```diff
@@ -177,34 +180,34 @@ const Spottable = hoc(defaultConfig, (config, Wrapped) => {
 			this.isHovered = false;
 
 			this.state = {
-				spottableDisabled: false
+				focusedWhenDisabled: false
 			};
 		}
 
+		componentDidMount () {
+			// eslint-disable-next-line react/no-find-dom-node
+			this.node = ReactDOM.findDOMNode(this);
+		}
+
 		componentWillReceiveProps (nextProps) {
-			const spottableDisabled = this.isFocused && nextProps.disabled || nextProps.spotlightDisabled;
+			const focusedWhenDisabled = this.isFocused && nextProps.disabled && !this.props.disabled;
 
-			if (!this.state.spottableDisabled && spottableDisabled) {
-				this.setState({spottableDisabled});
-			}
+			this.setState({
```

Comment:
Can you add the original guard back? Setting state to a value that may already exist will result in an unnecessary forced update. Seems like React could/should prevent against that ¯\_(ツ)_/¯

Reply:
There _shouldn't_ be any "extra" updates caused by setting state in `componentWillReceiveProps`. It will already be in the middle of an update cycle due to new props.

Reply:
The latest React versions will not update if `setState` is called with `null`, but I don't think that setting it to an existing state does the same thing.

Reply:
@webOS101 I don't believe that applies here. The handling of `setState` is different in terms of triggering re-renders when called within `componentWillReceiveProps` than elsewhere. Consider that its being migrated to a static method `getDerivedStateFromProps` (or whatever the name is). It's designed to generate state during a update cycle (initiated by a change of props).

Here's a quick example counting renders with a toggling `setState()` in `componentWillReceiveProps`.

https://codepen.io/ryanjduffy/pen/wjYmQw?editors=0010

Reply:
Sorry, it was a general comment, not specific to this case.  I understood that calling `setState` here causes no additional render.  I was speaking about calling it from elsewhere.